execution/process.go

```diff
@@ -310,22 +319,26 @@ func (p *ProcessByKey) Close(ctx context.Context, storage storage.Storage) error
 		return errors.Wrap(err, "couldn't stop garbage collector")
 	}
 
+	if err := storage.DropAll(outputWatermarkPrefix); err != nil {
+		return errors.Wrap(err, "couldn't clear storage with output watermark prefix")
+	}
+
 	return nil
 }
 
 type GarbageCollectorInfo struct {
-	garbageCollectorBoundary int // number of seconds subtracted from watermark to create boundary for garbage collection
-	garbageCollectorCycle    int // number of seconds for garbage collector to sleep between records check
+	garbageCollectorBoundary time.Duration // number of seconds subtracted from watermark to create boundary for garbage collection
+	garbageCollectorCycle    time.Duration // number of seconds for garbage collector to sleep between records check
 }
 
-func NewGarbageCollectorInfo(garbageCollectorBoundary, garbageCollectorCycle int) *GarbageCollectorInfo {
+func NewGarbageCollectorInfo(garbageCollectorBoundary, garbageCollectorCycle time.Duration) *GarbageCollectorInfo {
 	return &GarbageCollectorInfo{
 		garbageCollectorBoundary: garbageCollectorBoundary,
 		garbageCollectorCycle:    garbageCollectorCycle,
 	}
 }
 
-func (p *ProcessByKey) RunGarbageCollector(ctx context.Context, prefixedStorage storage.Storage, gbBoundary, gbCycle int) error {
+func (p *ProcessByKey) RunGarbageCollector(ctx context.Context, prefixedStorage storage.Storage, gbBoundary, gbCycle time.Duration) error {
 	for {
```

Comment:
Instead of the `sleep` at the very end maybe it's more elegant to do `for range time.Tick(gbCycle * time.Second)`

Reply:
absolutely right, completely forgot about this since i've never used tick before :P